src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -301,6 +308,19 @@ protected SCMRevision retrieve(SCMHead head, TaskListener listener)
                     ((BitbucketBranchSCMHead) head).getLatestCommit());
         }
 
+        if (head instanceof BitbucketTagSCMHead) {
+            if (((BitbucketTagSCMHead) head).getLatestCommit() == null) {
```

Comment:
Does the `BitbucketTagSCMHead` persist the latest commit in this code, and what happens if someone pushes up a change without a webhook, then tries to build this job?
I don't think we can trust this value- this is the "last _seen_ commit", not the latest commit. I have no opposition to persisting it, but wanted to see if we should have this conditional or always do the fetch.

Reply:
If we dont have a commit for the tag, fetch it, then push and build again without webhooks nothing will change. Tags are stuck on the commit they were created on. So we never have to fetch again and can trust it once we have a value.

Reply:
> Tags are stuck on the commit they were created on.

You can force push a different commit to a tag with the same name to Bitbucket. It's generally not great, but there are legitimate cases for doing so (burning a release as part of a `mvn release:prepare` comes immediately to mind).

I tried this out locally- make a branch with commits `a` and `b`, tag `a` as newtag and push & build it in Jenkins, then `git tag -f newtag` with `b` checked out, force push it up, and build again- Jenkins will continue to build `a`.

Tags _can_ change, but I don't think they change very often, and there's a good reason why Jenkins doesn't automatically build them so my gut feel is the slight inefficiency of checking every time, even though 99% of the time it will be the same is probably worth not having this issue- if you do want to rebuild a tag with a different commit, you either have to push up a different tag or edit it in UI which isn't great.

Reply:
All fixed.